dss/events/handlers/notify_v2.py

```diff
@@ -61,22 +61,24 @@ def notify_or_queue(replica: Replica, subscription: dict, metadata_document: dic
            not attempted for previously tombstoned versions. Since the number of versions is
            unbounded, inline delivery is not attempted.
     """
-    if _unversioned_tombstone_key_regex.match(key):
-        tombstones = set()
-        bundles = set()
-        key_prefix = key.rsplit(".", 1)[0]  # chop off the tombstone suffix
-        for key in _list_prefix(replica, key_prefix):
-            if _versioned_tombstone_key_regex.match(key):
-                bundle_key = key.rsplit(".", 1)[0]
-                tombstones.add(bundle_key)
-            elif _bundle_key_regex.match(key):
-                bundles.add(key)
-        for key in bundles:
-            if key not in tombstones:
-                queue_notification(replica, subscription, "TOMBSTONE", key, delay_seconds=0)
-    else:
-        if not notify(subscription, metadata_document, key):
-            queue_notification(replica, subscription, "CREATE", key)
+    with MessageQueuer(get_queue_url(notification_queue_name)) as mq:
+        if _unversioned_tombstone_key_regex.match(key):
+            tombstones = set()
+            bundles = set()
+            key_prefix = key.rsplit(".", 1)[0]  # chop off the tombstone suffix
+            for key in _list_prefix(replica, key_prefix):
+                if _versioned_tombstone_key_regex.match(key):
+                    bundle_key = key.rsplit(".", 1)[0]
+                    tombstones.add(bundle_key)
+                elif _bundle_key_regex.match(key):
+                    bundles.add(key)
+            for key in bundles:
+                if key not in tombstones:
+                    mq.send(_format_sqs_message(replica, subscription, "TOMBSTONE", key), delay_seconds=0)
+        else:
+            bundle_key = key.rsplit(".", 1)[0]
+            if not notify(subscription, metadata_document, bundle_key):
+                mq.send(_format_sqs_message(replica, subscription, "TOMBSTONE", bundle_key), delay_seconds=15 * 60)
```

Comment:
Maybe a dumb question, but why is this set to 15 minutes?

Reply:
I would preferred longer, but 15 minutes is the [maximum](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-timers.html)

Reply:
Good to know.